File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/ProjectRoleAssociation.java
package net.nemerosa.ontrack.model.security;

import lombok.Data;

import java.io.Serializable;

/**
 * Association of a project role and a projet ID.
 */
@Data
public class ProjectRoleAssociation implements Serializable {

    private final int projectId;
    private final ProjectRole projectRole;

    public boolean isGranted(Class<? extends ProjectFunction> fn) {
        return projectRole.isGranted(fn);
    }
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/ProjectRoleAssociation.kt
package net.nemerosa.ontrack.model.security

import java.io.Serializable

/**
 * Association of a project role and a projet ID.
 */
data class ProjectRoleAssociation(val projectId: Int, val projectRole: ProjectRole) : Serializable {

    fun isGranted(fn: Class<out ProjectFunction>): Boolean = projectRole.isGranted(fn)

}